_posts/2017-01-12-Node-with-TypeScript.md

```diff
@@ -0,0 +1,574 @@
+It also does not require any compilation, which makes running your code very easy. However this also removes any help from the compiler as no compile time checks are performed.
+No type checking, no checking for illogical structures or things that will just not work.
+<br/><br/>
+Code for Node can be run by simple opening a command prompt or terminal window and typing     
+    {% highlight shell %}
+    
+    node
+       
+    {% endhighlight %}
+This will start a Node instance and present you with an interpreter. You can now type commands and press return to execute them.
+This can be handy to test something quickly.
+<br/><br/>
+It is also possible to run a JavaScript file directly. This can be done via:
+    {% highlight shell %}
+    
+    node path/to/javascript-file.js 
+       
+    {% endhighlight %}
+<br/>
+However most of the time you will not be using this way of running code. Most of the time you will use npm to install your dependencies and start the node instance:
+    {% highlight shell %}
+    
+    npm install
+    npm start
+       
+    {% endhighlight %}
+This reads the package.json file and executes the scripts contained inside it.
+
+> Extensive documentation about the package.json file can be found on the [NPM website](https://docs.npmjs.com/files/package.json)
+
+### TypeScript you say!?
+
+<p style="text-align: center;">
+  <img alt="TypeScript" src="/img/node-with-typescript/typescript.jpg">
+</p>
+
+TypeScript has been around for some years now. It can be seen as a superset of JavaScript. It uses the same syntax but adds among other things compile time type checking. 
+It also adds a more Object Oriented model. A detailed explenation of the differences of the prototype based JavaScript and a more Object Oriented language can be found on the 
+[Mozilla Developer website](https://developer.mozilla.org/nl/docs/Web/JavaScript/Guide/Details_of_the_Object_Model)
+<br/><br/>
+
+[TypeScript developed mainly by Microsoft](https://www.typescriptlang.org/) and is completely open source! This means developers can make suggestion and report bugs (and even fix these bugs if they want).
+> TypeScript is a typed superset of JavaScript that compiles to plain JavaScript. Any browser. Any host. Any OS. Open source.
+
+<br/><br/>
+[TypeScript is very well documented](https://www.typescriptlang.org/docs/tutorial.html) and getting started with the language is fairly easy. 
+A lot of common development tools have support for TypeScript syntax checking. These include but are not limited to:
+- Intellij
+- Webstorm
+- Atom
+- Visual Studio Code
+- ...
+
+<br/><br/>
+As Node applications regularly use other NPM dependencies it is required for the TypeScript compiler to know about these dependencies and what types they use.
+You could make or generate these typings yourself, however you can easily find these typings on [TypeSearch](https://microsoft.github.io/TypeSearch/) website. 
```

Comment:
You could make or generate these typings yourself.**However,** you can easily find these typings on